### PR TITLE
[CARBONDATA-3925] flink write carbon file to hdfs when file size is less than 1M,can't write

### PR DESCRIPTION
**Pull request by yutaoChina**
### write carbon file less than 1M , less than configured minimum value (dfs.namenode.fs-limits.min-block-size) 1M ;can't write hdfs file
 

 
 ### the default block size is set  1024 byte and compare with  file size;when file size big than 1KB but less than 1M ,can not write file to hdfs ;so set default block size to 2M  

    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - No

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Can one of the admins verify this patch?

**Comment by ajantha-bhat** [comment]
Add to whitelist

**Comment by ajantha-bhat** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3741/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/2001/

**Comment by ajantha-bhat** [comment]
@yutaoChina : Thanks for working on this.
a) please handle the compilation error
b) please create a jira issue and add it in the issue header

File: integration/flink/src/main/java/org/apache/carbon/core/metadata/StageManager.java [hunk 1]
```diff
@@ -81,7 +81,7 @@ public static void writeStageInput(final String stageInputPath, final StageInput
   private static void writeSuccessFile(final String successFilePath) throws IOException {
     final DataOutputStream segmentStatusSuccessOutputStream =
         FileFactory.getDataOutputStream(successFilePath,
-            CarbonCommonConstants.BYTEBUFFER_SIZE, 1024);
+            CarbonCommonConstants.BYTEBUFFER_SIZE, 1024 * 1024 * 2);
```
Comment (ajantha-bhat) [inline integration/flink/src/main/java/org/apache/carbon/core/metadata/StageManager.java:84]: what if the file size is greater than 2 MB  ? why 2MB selected for this ?
may be need to pass the actual file size ?

Comment (yutaoChina) [reply integration/flink/src/main/java/org/apache/carbon/core/metadata/StageManager.java:84]: i set it 2M beacuase hdfs (dfs.namenode.fs-limits.min-block-size) configured minimum value size is 1M and in CarbonUtil.java class 
`getMaxOfBlockAndFileSize(long blockSize, long fileSize) `method use `long maxSize = blockSize;
    if (fileSize > blockSize) {
      maxSize = fileSize;
    }` 
if default size or filesize less than 1M program will get error ;
why 2M ? default is 1M so default * 2 bigger than it

**Comment by yutaoChina** [comment]
> 
> 
> @yutaoChina : Thanks for working on this.
> a) please handle the compilation error
> b) please create a jira issue and add it in the issue header
this is jira issues
https://issues.apache.org/jira/browse/CARBONDATA-3925

**Comment by ajantha-bhat** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3805/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/2064/

**Comment by yutaoChina** [comment]
> 
> 
> @yutaoChina : Thanks for working on this.
> a) please handle the compilation error
> b) please create a jira issue and add it in the issue header

my jira id is yutaochina

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/2068/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3809/

**Comment by ajantha-bhat** [comment]
@yutaoChina : you can run 
`mvn clean install -Pspark-2.3 -Pbuild-with-format -DskipTests` in your IDE to find all findbugs, checkstyle, scalastyle issues.
Now also it has some check style issue

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/2080/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3821/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/2086/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3827/

**Comment by ajantha-bhat** [comment]
LGTM. Thanks for the contribution!

Spark 2.3.4 build failure is a known random issue, we are working on it. So, merging this PR.

**Pull request closed without merging**